File: javascript/examples/post_payment_methods.js
const path = require('path');
process.chdir(path.resolve(__dirname, '../../'));
require('dotenv').config();

// Example of POST /payment-methods

const CARD_NUMBER = '4000056655665556'
const EXP_MONTH = '12'
const EXP_YEAR = '2028'
const CVC = '123'
const FIRST_NAME = 'Cookie'
const LAST_NAME = 'Monster'
const BILLING_ADDRESS= {
    "city": "New York",
    "state": "NY",
    "zip": "10463",
    "country": "US",
    "address1": "1234 Seasame Street",
    "address2": null
  }

let body = { 
	number: CARD_NUMBER,
	exp_month: EXP_MONTH,
	exp_year: EXP_YEAR,
	cvc: CVC,
	first_name: FIRST_NAME,
	last_name: LAST_NAME,
	billing_address: BILLING_ADDRESS
};

token = require('../request.js').request.token({ path: '/payment-methods', body: body });

require('../request.js').request.post({ path: '/payment-methods', token: token, body: body });

File: ruby/examples/post_payment_methods.rb
require_relative '../request.rb'

# Example POST /payment-methods

CARD_NUMBER = '6011111111111117'
EXP_MONTH = '12'
EXP_YEAR = '2028'
CVC = '456'
FIRST_NAME = 'Ernie'
LAST_NAME = 'Bert'
BILLING_ADDRESS= {
    "city": "New York",
    "state": "NY",
    "zip": "10463",
    "country": "US",
    "address1": "1415 Seasame Street",
    "address2": nil
  }

body = { 
	number: CARD_NUMBER,
	exp_month: EXP_MONTH,
	exp_year: EXP_YEAR,
	cvc: CVC,
	first_name: FIRST_NAME,
	last_name: LAST_NAME,
	billing_address: BILLING_ADDRESS
}.to_json

BaseRequestRoute.request(:post, '/payment-methods', body: body)